### PR TITLE
Feature 2fac pairing

### PR DESCRIPTION
**Pull request by SSpedsbjerg**

### CODE REVIEW

File: src/main/main.c [hunk 1]
```diff
@@ -49,6 +50,91 @@ bool open_button = false;
...
+} Bluetooth_address;
+
+Bluetooth_address authenticated_units[10];
+int auth_iterator = 0;
+
+Bluetooth_address lastest_bluetooth_unit;
+
+//returns true if pressed and adds the unit to the authenticated_units array, false if not
+bool update_auth() {
+    if (gpio_get_level(PIN_AUTH) == 0) { //pressed
```
Comment (vimian) [inline src/main/main.c]: Should be moved together with the other buttons in listen_to_buttons()

File: src/main/main.c [hunk 2]
```diff
@@ -89,7 +165,7 @@ void opened() {
 void func_unlocking() {
     if (unlock_timer == 10 && !is_opened) {
         state = LOCKED;
-    } else if (unlock_timer < 10 && !is_opened) {
+    } else if (unlock_timer < 10 && !is_opened && is_authenticated()) {
```
Comment (vimian) [inline src/main/main.c]: move this to bt connection

if (is_locked && !unlocking && !manual_lock && is_authenticated()) {
                unlocking = true;
                unlock_timer = 0;

**Review by vimian — APPROVED** [review]

File: src/main/main.c [hunk 3]
```diff
@@ -319,6 +418,8 @@ void app_main(void) {
 
     setup_bt();
 
+
```
Comment (GachiRoss) [inline src/main/main.c:421]: unecesary lines

**Review by GachiRoss — CHANGES_REQUESTED** [review]
reason for adding lines ?

File: src/main/main.c (hunk 3 above)
Comment (vimian) [reply src/main/main.c:421]: It is okay

**Review by GachiRoss — APPROVED** [review]

**Pull request merged**